zellij-server/src/panes/grid.rs

```diff
@@ -1052,20 +1052,21 @@ impl Grid {
                 .selection
                 .contains_row(character_chunk.y.saturating_sub(content_y))
             {
-                let background_color = match style.colors.bg {
-                    PaletteColor::Rgb(rgb) => AnsiCode::RgbCode(rgb),
-                    PaletteColor::EightBit(col) => AnsiCode::ColorIndex(col),
+                let selection = self.selection.offset(content_x, content_y);
```

Comment:
I've moved the selection offsetting here as depending on `offset_*` in `add_selection_and_colors` or now `SelectionColors::invert / discrete` seems like a layer violation. Those methods expect a `Selection` for determining which color styles to apply, but we shouldn't leak another implementation detail into them. 

However, this doesn't eliminate the risk of forgetting to call `Selection::offset`. Maybe returning an `OffsetSelection` newtype from `Selection::offset` and depending on that in `SelectionColors` would ensure this guarantee.

Reply:
I've added this guarantee in  54c107af. Now the compiler will complain if you try to pass a `Selection` to `SelectionColors`, so we _must_ call `offset` first.